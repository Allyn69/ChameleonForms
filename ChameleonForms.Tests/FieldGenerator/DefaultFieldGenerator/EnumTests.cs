using System.Collections.Generic;
using ApprovalTests.Html;
using ChameleonForms.Component.Config;
using ChameleonForms.Templates;
using NUnit.Framework;

namespace ChameleonForms.Tests.FieldGenerator.DefaultFieldGenerator
{
    class EnumTests : DefaultFieldGeneratorShould
    {
        [Test]
        public void Use_correct_html_for_required_enum_field()
        {
            var g = Arrange(m => m.RequiredEnum, m => m.RequiredEnum = TestEnum.ValueWithDescriptionAttribute);

            var result = g.GetFieldHtml(new FieldConfiguration { Attributes = new HtmlAttributes(data_attr => "value") });

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_optional_enum_field()
        {
            var g = Arrange(m => m.OptionalEnum, m => m.OptionalEnum = null);

            var result = g.GetFieldHtml(new FieldConfiguration { Attributes = new HtmlAttributes(data_attr => "value") });

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_nullable_required_enum_field()
        {
            var g = Arrange(m => m.RequiredNullableEnum);

            var result = g.GetFieldHtml(new FieldConfiguration { Attributes = new HtmlAttributes(data_attr => "value") });

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_required_enum_list_field()
        {
            var g = Arrange(m => m.RequiredEnumList, m => m.RequiredEnumList = new List<TestEnum>{TestEnum.Simplevalue,TestEnum.ValueWithDescriptionAttribute});

            var result = g.GetFieldHtml(new FieldConfiguration { Attributes = new HtmlAttributes(data_attr => "value") }.AsList());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_required_nullable_enum_list_field()
        {
            var g = Arrange(m => m.RequiredNullableEnumList);

            var result = g.GetFieldHtml(new FieldConfiguration { Attributes = new HtmlAttributes(data_attr => "value") });

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_optional_enum_list_field()
        {
            var g = Arrange(m => m.OptionalEnumList);

            var result = g.GetFieldHtml(new FieldConfiguration { Attributes = new HtmlAttributes(data_attr => "value") });

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_optional_nullable_enum_list_field()
        {
            var g = Arrange(m => m.OptionalNullableEnumList, m => m.OptionalNullableEnumList = new List<TestEnum?> { TestEnum.Simplevalue, TestEnum.ValueWithDescriptionAttribute });

            var result = g.GetFieldHtml(new FieldConfiguration { Attributes = new HtmlAttributes(data_attr => "value") }.AsList());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }
    }
}
