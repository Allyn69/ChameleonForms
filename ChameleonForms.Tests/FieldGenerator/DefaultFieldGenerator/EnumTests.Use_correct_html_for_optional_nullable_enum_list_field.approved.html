<ul>
  <li>
    <input checked="checked" class="input-validation-error" data-attr="value" id="OptionalNullableEnumList_1" name="OptionalNullableEnumList" type="checkbox" value="Simplevalue" />
    <label for="OptionalNullableEnumList_1">Simplevalue</label>
  </li>
  <li>
    <input checked="checked" class="input-validation-error" data-attr="value" id="OptionalNullableEnumList_2" name="OptionalNullableEnumList" type="checkbox" value="ValueWithDescriptionAttribute" />
    <label for="OptionalNullableEnumList_2">Description attr text</label>
  </li>
  <li>
    <input class="input-validation-error" data-attr="value" id="OptionalNullableEnumList_3" name="OptionalNullableEnumList" type="checkbox" value="ValueWithMultpipleWordsAndNoDescriptionAttribute" />
    <label for="OptionalNullableEnumList_3">Value with multpiple words and no description attribute</label>
  </li>
</ul>